src/compiler/types.ts

```diff
@@ -5394,7 +5404,7 @@ export const enum NodeBuilderFlags {
     AllowUniqueESSymbolType                 = 1 << 20,
     AllowEmptyIndexInfoType                 = 1 << 21,
     /** @internal */ WriteComputedProps      = 1 << 30, // { [E.A]: 1 }
-
+    /** @internal */ NoSyntacticPrinter     = 1 << 31,

```

Comment:
When the syntactic node builder will do more than just emit errors (hopefully very very soon) there will not be a need for this as the syntactic node builder should be the one building the types (and if it fails falls back to inference). But for now we need a way to avoid checking the nodes at multiple levels (ex: checking the whole object literal, then also checking the properties individually as they are added to the synthetic object type node)